third_party/opengrep/install_opengrep.py

```diff
@@ -0,0 +1,182 @@
+    if system == 'Darwin':
+        if arch == 'arm64':
+            return 'opengrep_osx_arm64', 'opengrep'
+        return 'opengrep_osx_x86', 'opengrep'
+
+    if system == 'Linux':
+        # Simple detection: assume glibc unless Alpine or similar
+        if arch == 'arm64':
+            return 'opengrep_manylinux_aarch64', 'opengrep'
+        return 'opengrep_manylinux_x86', 'opengrep'
+
+    if system == 'Windows':
+        if arch == 'arm64':
+            print('Windows ARM64 is not supported')
+            return None, None
+        return 'opengrep_windows_x86.exe', 'opengrep.exe'
+
+    print(f'Unsupported operating system: {system}')
+    return None, None
+
+
+def IsOpengrepInstalled(binary_name):
+    """Check if opengrep is already installed with the correct version."""
+    binary_path = os.path.join(INSTALL_DIR, binary_name)
+    version_file = os.path.join(INSTALL_DIR, '.version')
+
+    if not os.path.exists(binary_path) or not os.path.exists(version_file):
+        return False
+
+    try:
+        with open(version_file, 'r') as f:
+            installed_version = f.read().strip()
+            if installed_version == OPENGREP_VERSION:
+                print(f'Opengrep {OPENGREP_VERSION} already installed')
+                return True
+    except IOError:
+        pass
+
+    return False
+
+
+def VerifyChecksum(file_path, expected_checksum):
+    """Verify SHA256 checksum of downloaded file."""
+    sha256 = hashlib.sha256()
+    with open(file_path, 'rb') as f:
+        for chunk in iter(lambda: f.read(4096), b''):
+            sha256.update(chunk)
+
+    actual = sha256.hexdigest()
+    if actual != expected_checksum:
+        raise RuntimeError(
+            f'Checksum mismatch!\nExpected: {expected_checksum}\nActual: {actual}'
+        )
+
+
+def InstallOpengrep():
+    """Download and install opengrep binary."""
+    dist_name, binary_name = GetPlatformDistName()
+    if not dist_name:
+        return 1
+
+    if IsOpengrepInstalled(binary_name):
+        return 0
+
+    binary_path = os.path.join(INSTALL_DIR, binary_name)
+    version_file = os.path.join(INSTALL_DIR, '.version')
+
+    # Construct download URL
+    url = (f'https://github.com/opengrep/opengrep/releases/download/'
+           f'{OPENGREP_VERSION}/{dist_name}')
+
+    print(f'Installing opengrep {OPENGREP_VERSION}...')
+
+    # Create installation directory
+    deps.EnsureDirExists(INSTALL_DIR)
+
+    # Download to temporary file
+    try:
+        import tempfile
+        with tempfile.NamedTemporaryFile(delete=False, dir=INSTALL_DIR) as tmp:
+            deps.DownloadUrl(url, tmp)
+            tmp_path = tmp.name
+
+        # Verify checksum
+        expected_checksum = BINARY_CHECKSUMS.get(dist_name)
+        if expected_checksum:
+            print('Verifying checksum...')
+            VerifyChecksum(tmp_path, expected_checksum)
+        else:
+            print(
+                'Warning: No checksum available for verification. Proceeding anyway.'
```

Comment:
why not raise?

Reply:
fixed in https://github.com/brave/brave-core/pull/32195/commits/34f74a24f2a4c8a883a5d1426a1c904cfb89d42b